### ISSUE TITLE
DSM 7.2.1-69057 Won't support syno_hdd_db.sh

### ISSUE THREAD

**Issue by hari-bo** [007revad/Synology_HDD_db#150]
I was running the following M2 in an RAID-1 Volume Kingston NV2 2 TB SSD (SNV2S/2000G) with DSM 7.2-64570-2.
But after upgrading to DSM 7.2.1-69057 the devices aren't detected anymore.

With an SSH session I've runned manually syno_hdd_db.sh -nfr but it won't detect the M2 storage anymore, it seems dat Synology has changed something which prevent non-Synology M2 storage to be used.

```
HDD/SSD models found: 1
MG07ACA12TE,0103

No M.2 drives found

No M.2 PCIe cards found

No Expansion Units found

Backed up ds918+_host_v7.db
Updated MG07ACA12TE to ds918+_host_v7.db

Backed up synoinfo.conf

Disabled support disk compatibility.

Set max memory to 16384 GB.

Disabled drive db auto updates.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.
```

I've tried several reboots and DSM reinstall (rollback to previous version isn't possible without wiping the complete disks).

**Comment by hari-bo** [comment]
If I can provide more information just let me know.

**Comment by 007revad** [comment]
Which Synology model do you have? Is it a real Synology NAS or Xpenology?

I upgraded from DSM 7.2-64570 to 7.2.1 and my M2 volume survived.

I also managed to rollback to 7.2-64570 without losing any data. See https://github.com/007revad/Synology_HDD_db/issues/149#issuecomment-1746325839

**Comment by hari-bo** [comment]
> Which Synology model do you have? Is it a real Synology NAS or Xpenology?
> 
> I upgraded from DSM 7.2-64570 to 7.2.1 and my M2 volume survived.
> 
> I also managed to rollback to 7.2-64570 without losing any data. See [#149 (comment)](https://github.com/007revad/Synology_HDD_db/issues/149#issuecomment-1746325839)

I’ve a DS918+ no Xpenogy.

**Comment by 007revad** [comment]
What do each of the following commands return:

`sudo -i nvme list`

`ls /run/synostorage/disks`

`ls /run/synostorage/disks/nvme0n1`

`ls /run/synostorage/disks/nvme1n1`

`get_key_value /etc.defaults/synoinfo.conf support_m2_pool`

`get_key_value /etc/synoinfo.conf support_m2_pool`

`get_key_value /etc.defaults/synoinfo.conf supportnvme`

`get_key_value /etc/synoinfo.conf supportnvme`

**Comment by hari-bo** [comment]
Here's the requested output.

```
sudo -i nvme list
empty

ls /run/synostorage/disks
sda  sdb  sdc (SATA Storage)

ls /run/synostorage/disks/nvme0n1
ls: cannot access '/run/synostorage/disks/nvme0n1': No such file or directory

ls /run/synostorage/disks/nvme1n1
ls: cannot access '/run/synostorage/disks/nvme1n1': No such file or directory

get_key_value /etc.defaults/synoinfo.conf support_m2_pool
no

get_key_value /etc/synoinfo.conf support_m2_pool
no

get_key_value /etc.defaults/synoinfo.conf supportnvme
yes

get_key_value /etc/synoinfo.conf supportnvme
yes
```

And i've got Home Assistant with Synology intergration and the M2 is then visible, but within the Storage Manager from DSM not.
![Screenshot at Oct 05 12-00-15](https://github.com/007revad/Synology_HDD_db/assets/2697468/29d88ae2-e0e5-49d8-be50-bead036e3a5b)

**Comment by 007revad** [comment]
Can you run this script and see if that makes the NVMe drive show in storage manager.
https://github.com/AuxXxilium/arc-extensions/blob/main/nvmecache/all/usr/sbin/nvmecache.sh

![download_button](https://github.com/007revad/Synology_enable_M2_volume/assets/39733752/f550113a-9d7d-43c2-993a-b1c180a6181e)

If that works can you then run syno_hdd_db.sh again.

**Comment by hari-bo** [comment]
I've runned the script under root user (it gives no output) but within the GUI and running syno_hdd_db.sh shows no M2 device yet.

**Comment by 007revad** [comment]
> I've runned the script under root user (it gives no output

It should have output something like this:
`nvme0 - 0000:00:14.0 - 303030303a30303a31342e30`

The only reason it would output nothing is if `ls /sys/class/nvme` returned nothing.

What do these commands return:

`ls /sys/class/nvme`

`ls /sys/block | grep nvme`

I wonder if Synology have blacklisted kingston M2 drives somewhere in DSM 7.2.1.  I installed 7.2.1 on my DS720+ and still shows my WD NVMe drive in storage manager. After I ran my syno_enable_m2_volume.sh script I was able to create a volume in storage manager.

Do you want to try https://github.com/007revad/Synology_enable_M2_volume

**Comment by hari-bo** [comment]
There's no ouput when requesting the information.
I've runned the script which shows an reboot messaged, after that no M2 storage visible.

**Comment by hari-bo** [comment]
Everything is fine.

**Comment by leginag** [comment]
Just adding that I had to follow the "physically remove M2 drives" instructions above to fix it on my 920+ when updating to 7.2.1
